base/src/main/java/com/smartdevicelink/managers/screen/menu/BaseMenuManager.java

```diff
@@ -248,1334 +202,255 @@ public void onComplete(Map<String, String> errors) {
      * @return a List of Currently set menu cells
      */
     public List<MenuCell> getMenuCells() {
-
-        if (menuCells != null) {
-            return menuCells;
-        } else if (waitingOnHMIUpdate && waitingUpdateMenuCells != null) {
-            // this will keep from returning null if the menu list is set and we are pending HMI FULL
-            return waitingUpdateMenuCells;
-        } else {
-            return null;
-        }
-    }
-
-    /**
-     * @return The currently set DynamicMenuUpdatesMode. It defaults to ON_WITH_COMPAT_MODE
-     */
-    public DynamicMenuUpdatesMode getDynamicMenuUpdatesMode() {
-        return this.dynamicMenuUpdatesMode;
+        return menuCells;
     }
 
-    // OPEN MENU RPCs
+    private boolean openMenuPrivate(MenuCell cell) {
+        MenuCell foundClonedCell = null;
 
-    /**
-     * Opens the Main Menu
-     */
-    public boolean openMenu() {
+        if (cell != null) {
+            // We must see if we have a copy of this cell, since we clone the objects
+            for (MenuCell clonedCell : currentMenuCells) {
+                if (clonedCell.equals(cell) && clonedCell.getCellId() != parentIdNotFound) {
+                    // We've found the correct sub menu cell
+                    foundClonedCell = clonedCell;
+                    break;
+                }
+            }
+        }
 
-        if (sdlMsgVersion.getMajorVersion() < 6) {
-            DebugTool.logWarning(TAG, "Menu opening is only supported on head units with RPC spec version 6.0.0 or later. Currently connected head unit RPC spec version is: " + sdlMsgVersion.getMajorVersion() + "." + sdlMsgVersion.getMinorVersion() + "." + sdlMsgVersion.getPatchVersion());
+        if (cell != null && (!cell.isSubMenuCell())) {
+            DebugTool.logError(TAG, String.format("The cell %s does not contain any sub cells, so no submenu can be opened", cell.getTitle()));
+            return false;
+        } else if (cell != null && foundClonedCell == null) {
+            DebugTool.logError(TAG, "This cell has not been sent to the head unit, so no submenu can be opened. Make sure that the cell exists in the SDLManager.menu array");
```

Comment:
List or ArrayList, I'm fine with either.
```suggestion
            DebugTool.logError(TAG, "This cell has not been sent to the head unit, so no submenu can be opened. Make sure that the cell exists in the SDLManager.menu list");
```